src/tribler/gui/core_manager.py

```diff
@@ -277,23 +277,36 @@ def get_last_core_output(self, quoted=True):
     @staticmethod
     def error_code_to_hex(error_code: int) -> str:
         """Convert a signed integer error code to a hexadecimal string."""
-        v = error_code & (2 ** 32 - 1)
+        v = error_code & 0xffffffff
         return hex(v)
 
+    @staticmethod
+    def format_str_error(error_code: int):
+        """ Return the error message corresponding to the given error code."""
+        try:
+            return os.strerror(error_code)
+        except ValueError:
+            # On platforms where strerror() returns NULL when given an unknown error number, ValueError is raised.
+            return '<no description>'
+
+    @staticmethod
+    def format_win_error(error_code: int):
+        """ Return the Windows error message corresponding to the given error code."""
+        import ctypes  # pylint: disable=import-outside-toplevel
+        return ctypes.FormatError(error_code)
```

Comment:
Unfortunately, from what I see, `ctypes.FormatError` does not return correct error descriptions for the actual exit codes we have. Based on your PR, I implemented a different idea: I found the complete list of Windows errors, parsed it and added a new utility function to retrieve the correct error name. Please look at #7885; it may be an approach that we should choose.